website/docs/sdk-reference/dotnet.md

```diff
@@ -240,19 +240,66 @@ User userObject = new User("john@example.com");
-    Custom = new Dictionary<string, string> {
-        {"SubscriptionType", "Pro"},
-        {"UserRole", "Admin"}}
+    Custom =
+    {
+        ["SubscriptionType"] = "Pro",
+        ["UserRole"] = "Admin"
+    }
+};
+```
+
+The `Custom` dictionary also allows attribute values other than `string` values:
+
+```csharp
+User userObject = new User("#UNIQUE-USER-IDENTIFIER#")
+{
+    Custom =
+    {
+        ["Rating"] = 4.5,
+        ["RegisteredAt"] = DateTimeOffset.Parse("2023-11-22 12:34:56 +00:00", CultureInfo.InvariantCulture),
+        ["Roles"] = new[] { "Role1", "Role2" }
+    }
 };
 ```
 
+The set of allowed attribute values depends on the comparison type of the condition which references the User Object attribute.
+`string` values are supported by all comparison types (in some cases they need to be provided in a specific format though).
+Some of the comparison types work with other types of values, as described below.
+
+**Text-based comparisons** (EQUALS, IS ONE OF, etc.)
+* accept `string` values,
+* all other values are automatically converted to `string` (a warning will be logged but evaluation will continue as normal).
+
+**SemVer-based comparisons** (IS ONE OF, &lt;, &gt;=, etc.)
+* accept `string` values containing a properly formatted, valid semver value,
+* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).
+
+**Number-based comparisons** (=, &lt;, &gt;=, etc.)
+* accept `double` values and all other numeric values which can safely be converted to `double`,
+* accept `string` values containing a properly formatted, valid `double` value,
+* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).
+  
+**Date time-based comparisons** (BEFORE / AFTER - available only in the new config model, *Config V2*)
+* accept `DateTime` or `DateTimeOffset` values, which are automatically converted to a second-based Unix timestamp,
+* accept `double` values representing a second-based Unix timestamp and all other numeric values which can safely be converted to `double`,
+* accept `string` values containing a properly formatted, valid `double` value,
+* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).
+  
+**String array-based comparisons** (ARRAY CONTAINS ANY OF / ARRAY NOT CONTAINS ANY OF - available only in the new config model, *Config V2*)
+* accept arrays of `string`,
+* accept `string` values containing a valid JSON string which can be deserialized to an array of `string`,
+* all other values are considered invalid (a warning will be logged and the currently evaluated targeting rule will be skipped).
+
+It is also worth mentioning that in case a non-string attribute value needs to be converted to `string` during evaluation,
+it will always be done using the same format which is accepted by the comparisons.
```

Comment:
Also unclear about what's going to happen.

Reply:
So, in cases described above an automatic conversion will happen (from whatever type to string). When it's an allowed value in user comparisons, the SDK will convert to the same string format which is accepted by the comparisons.

For example, when the custom attribute is a `Date` object, it will be converted to a string containing a number which represent the equivalent Unix timestamp.